pkg/lifecycle/service.go

```diff
@@ -176,6 +211,44 @@ func (s *Service) Start(
 	return nil
 }
 
+// StartWithBackoff starts a pipeline with a backoff.
+// It'll check the number of times the pipeline has been restarted and the duration of the backoff.
+// When the pipeline has reached out the maximum number of retries, it'll return a fatal error.
+func (s *Service) StartWithBackoff(ctx context.Context, rp *runnablePipeline) error {
+	s.logger.Info(ctx).Str(log.PipelineIDField, rp.pipeline.ID).Msg("restarting with backoff")
```

Comment:
@raulb FYI, changed this, because a restart might be a significant event. I'm open to changing back if you think it's too high of a logging level.

Reply:
@hariso makes sense!